postgres-appliance/upload_pg_log_to_s3.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import boto
+import math
+import os
+import logging
+import gzip
+import shutil
+import time
+
+from datetime import datetime, timedelta
+from filechunkio import FileChunkIO
+
+
+def compress_pg_log():
+
+    yesterday = datetime.now() - timedelta(days=1)
+    yesterday_day_number = yesterday.strftime('%u')
+
+    log_file = os.getenv('PGLOG') + "/postgresql-" + yesterday_day_number + ".csv"
+    archived_log_file = os.getenv('LOG_TMPDIR') + "/" + yesterday.strftime('%Y-%m-%d') + ".csv.gz"
+
+    if os.path.getsize(log_file) == 0:
+        logging.warn("Postgres log from yesterday '%s' is empty.", log_file)
+        exit(0)
+
+    with open(log_file, 'rb') as f_in:
+        with gzip.open(archived_log_file, 'wb') as f_out:
+            shutil.copyfileobj(f_in, f_out)
+
+    return archived_log_file
+
+
+def upload_to_s3(local_file_path):
+
+    # boto picks up AWS credentials automatically when run within a EC2 instance
+
+    # host sets a region and the correct AWS SignatureVersion along the way
+    # see https://github.com/boto/boto/issues/2741
+    conn = boto.connect_s3(host=os.getenv('LOG_AWS_HOST'))
+
+    bucket_name = os.getenv('LOG_S3_BUCKET')
+    bucket = conn.get_bucket(bucket_name, validate=False)
+
+    key_name = os.getenv('LOG_S3_KEY') + '/' + os.path.basename(local_file_path)
+    mp_upload = bucket.initiate_multipart_upload(key_name)
+
+    chunk_size = 52428800  # 50 MiB
+    file_size = os.stat(local_file_path).st_size
+    chunk_count = int(math.ceil(file_size / float(chunk_size)))
+
+    for i in range(chunk_count):
+        offset = chunk_size * i
+        bytes = min(chunk_size, file_size - offset)
+        with FileChunkIO(local_file_path, 'r', offset=offset, bytes=bytes) as fp:
+            try:
+                mp_upload.upload_part_from_file(fp, part_num=i + 1)
+            except Exception as e:
+                logging.critical(
+                    "Failed to upload the {} to the bucket {} under the key {}. Cancelling the multipart upload {}: {}"
+                    .format(local_file_path, mp_upload.bucket_name, mp_upload.key_name, mp_upload.id, e))
+                cancel_multipart_upload(mp_upload)
+                exit(1)
```

Comment:
Basically that means we will lose logs for one day if it fails.
Why there is no retry?